src/test/java/edu/cornell/kfs/fp/document/service/impl/CuDisbursementVoucherPaymentReasonServiceImplTest.java

```diff
@@ -1,85 +1,107 @@
+        disbursementVoucherPaymentReasonService = new TestableDisbursementVoucherPaymentReasonServiceImpl();
+        disbursementVoucherPaymentReasonService.setDisbursementVoucherPayeeService(buildMockDisbursementVoucherPayeeService());
+        cuDisbursementVoucherPaymentReasonService = new CuDisbursementVoucherPaymentReasonServiceImpl();
+        cuDisbursementVoucherPaymentReasonService.setDisbursementVoucherPayeeService(buildMockDisbursementVoucherPayeeService());
+    }
+    
+    @After
+    public void tearDown() {
+        disbursementVoucherPaymentReasonService = null;
+        disbursementVoucherPaymentReasonService = null;
+        payeeTypeCodeMap = null;
+    }
+
+    private void populatePayeetypeCodeMap() {
+        payeeTypeCodeMap = new HashMap<String, String>();
+        payeeTypeCodeMap.put("E", "Employees Students Alumni");
+        payeeTypeCodeMap.put("S", "Employees Students Alumni");
+        payeeTypeCodeMap.put("A", "Employees Students Alumni");
+        payeeTypeCodeMap.put("V", "Vendor");
+    }
+
+    private DisbursementVoucherPayeeService buildMockDisbursementVoucherPayeeService() {
+        DisbursementVoucherPayeeService payeeService = Mockito.mock(DisbursementVoucherPayeeService.class);
+        Mockito.when(payeeService.getPayeeTypeDescription(Mockito.any())).then(this::findPayeeTypeDescription);
+        return payeeService;
+    }
+    
+    private String findPayeeTypeDescription(InvocationOnMock invocation) {
+        String payeeTypeCode = invocation.getArgument(0);
+        if (payeeTypeCodeMap.containsKey(payeeTypeCode)) {
+            return payeeTypeCodeMap.get(payeeTypeCode);
+        }
+        return StringUtils.EMPTY;
+    }
     
+    @Test
+    public void testGetDescriptivePayeeTypesAsString() {
+        Collection<String> payeeTypeCds = buildPayeeTypeCodeCollection();
+
+        String eduResult = cuDisbursementVoucherPaymentReasonService.getDescriptivePayeeTypesAsString(payeeTypeCds);
+        String orgResult = disbursementVoucherPaymentReasonService.getDescriptivePayeeTypesAsString(payeeTypeCds);
+
+        if (eduResult.toString().length() >= orgResult.toString().length()) {
+            fail("The edu method should return a smaller string");
+        }
+
+    }
+
+    protected Collection<String> buildPayeeTypeCodeCollection() {
+        Collection<String> payeeTypeCds = new ArrayList<String>();
+        payeeTypeCds.add("E");
+        payeeTypeCds.add("F");
+        payeeTypeCds.add("T");
+        payeeTypeCds.add("V");
+        payeeTypeCds.add("E");
+        payeeTypeCds.add("F");
+        payeeTypeCds.add("X");
+        payeeTypeCds.add("S");
+        return payeeTypeCds;
+    }
+
+    public class TestableDisbursementVoucherPaymentReasonServiceImpl extends DisbursementVoucherPaymentReasonServiceImpl {
+
+        /*
+         * This is the same function as is in DisbursementVoucherPaymentReasonServiceImpl, except it doesn't use SpringContext
+         */
```

Comment:
Is this comment necessary?

Reply:
I thought it would be helpful, but I can remove it